src/Education/XamlGenerator/RuleHelpXamlTranslator.cs

```diff
@@ -356,9 +357,34 @@ private void WriteText(string text)
             // e.g. <li> some text ... </li>            -> <li> does not support inlines directly.
             EnsureCurrentOutputSupportsInlines();
 
-            // Note: we could explicitly wrap the text in a <Run>. However, that will happen implicitly
-            // when the XAML is parsed, and it won't make any difference to the rendered output.
-            writer.WriteString(text);
+            // Parse text to see if it contains any cross reference rules as they the need to be treated as links.
+            var textTokens = CustomTextMarkupParser.Parse(text);
+
+            foreach (var textToken in textTokens)
+            {
+                if (textToken is ISimpleText simpleText)
+                {
+                    // Note: we could explicitly wrap the text in a <Run>. However, that will happen implicitly
+                    // when the XAML is parsed, and it won't make any difference to the rendered output.
+                    writer.WriteString(simpleText.Text);
+                }
+                else if (textToken is IRuleCrossRef ruleCrossRef)
+                {
+                    HandleCrossReference(ruleCrossRef.CompositeRuleId);
+                }
+
+                Debug.Assert(textToken is IRuleCrossRef || textToken is ISimpleText, "Unknown text token type.");
```

Comment:
It's always worth putting as much info into the assert message as possible e.g.

`$"Unknown text token type. Type: {textToken.GetType().FullName}"`

Reply:
About this, I just realized should I have a test that covers this happening? So return a textToken thats neither of these types and see if it debug asserts?

Reply:
We don't need to check that the assertion happens since the assert is just a debug-time helper for us, and we don't want to have to write tests for debug-specific behaviour.

However, you could write a test to check the expected product code behaviour in this case i.e. that unknown types are ignored (you'd need to use our `AssertIgnoreScope` class in the test to stop the assertion from popping up a dialogue and interrupting the test).